dashboard/lib/services/teacher.rb

```diff
@@ -0,0 +1,15 @@
+module Services
+  module Teacher
+    # Updates the user permission to AUTHORIZED_TEACHER
+    #
+    # @param [User]
+    def self.verify(teacher)
```

Comment:
Should this be `verify!`? It is saving the record and has some side effects, such as sending a verification email, which could justify an exclamation mark.

Reply:
If I understood correctly, I did not use the exclamation in the name because it returns a true/false and does not raise an error.

> `save!` will raise an error if not successful.
> `save` will return boolean value like true or false.

Reply:
The method itself isn't "dangerous":
- https://thecodest.co/blog/ruby-software-development-bang-methods-and-ways-to-create-them/#:~:text=Bang%20methods%2C%20denoted%20by%20an,without%20creating%20a%20separate%20copy.
- https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#dangerous-method-suffix

Reply:
Ok, maybe we can add a comment that warns that verifying a teacher will send an email.